### DISCUSSION TITLE
PMD stderr:Error: Could not find or load main class net.sourceforge.pmd.PMD

### DISCUSSION THREAD

**Discussion by Sandhya-Rakesh** [pmd/pmd#3623] [Q&A] [answered]
Hi All,

I am new to PMD and trying to use it for Apex. I have installed PMD and Integrated it with VS code. I have installed all the Java requirements as well. But I am receiving the following error. Am I missing any other configurations? 

```
PMD Command: java -cp "C:\EntsysTools\pmd-bin-6.40.0\bin\lib\*;c:\Sandhya\SalesforceVSProjects\Rotman\*" net.sourceforge.pmd.PMD -f csv -no-cache -d "c:\Sandhya\SalesforceVSProjects\Rotman\force-app\main\default\classes\RefundController.cls" -R "c:\Users\srakesh\.vscode\extensions\chuckjonas.apex-pmd-0.5.6\rulesets\apex_ruleset.xml"
stderr:Error: Could not find or load main class net.sourceforge.pmd.PMD

```

**Comment by adangel** [comment]
The classpath is wrong, there is one `bin` too much, the `lib` directory is at the same level as bin.... So your command that should be executed, should look like:

```
PMD Command: java -cp "C:\EntsysTools\pmd-bin-6.40.0\lib\*;c:\Sandhya\SalesforceVSProjects\Rotman\*" net.sourceforge.pmd.PMD -f csv -no-cache -d "c:\Sandhya\SalesforceVSProjects\Rotman\force-app\main\default\classes\RefundController.cls" -R "c:\Users\srakesh\.vscode\extensions\chuckjonas.apex-pmd-0.5.6\rulesets\apex_ruleset.xml"
```

Don't know where or how you configure this. Maybe you selected the "bin" folder instead of the base-folder where you extracted PMD?

**Comment by adangel** [comment, marked as answer]
Are you using this plugin? https://marketplace.visualstudio.com/items?itemName=chuckjonas.apex-pmd

The you need to configure it as it is described:

> pmdBinPath (prev. pmdPath) (optional): set to override the default pmd binaries. This should point to the PMD folder which contains folders lib and bin. Most likely it is called libexec.

You seem to have selected the bin folder instead of the folder "which contains folders lib and bin".
